Algo/src/roots.cxx

```diff
@@ -16,10 +16,11 @@ namespace Algo {
 using Arith::mat, Arith::vec;
 
 namespace {
-template <typename T> vec<T, 2> gn(vec<T, 2> x) {
+template <typename T> CCTK_HOST CCTK_DEVICE vec<T, 2> gn(vec<T, 2> x) {
```

Comment:
If these functions are only called inside a kernel then the `CCTK_HOST` declaration is not necessary.